EcoSoapBank/Pickups/NewPickupViewController.swift

```diff
@@ -0,0 +1,263 @@
+//
+//  NewPickupViewController.swift
+//  EcoSoapBank
+//
+//  Created by Jon Bash on 2020-08-17.
+//  Copyright © 2020 Spencer Curtis. All rights reserved.
+//
+
+import UIKit
+import SwiftUI
+import Combine
+
+
+class NewPickupViewController: UIViewController {
+    typealias Snapshot = NSDiffableDataSourceSnapshot<Int, NewCartonViewModel>
+
+    private var viewModel: NewPickupViewModel
+    private var cancellables: Set<AnyCancellable> = []
+
+    private lazy var tableViewHeight: NSLayoutConstraint =
+        tableView.heightAnchor.constraint(equalToConstant: 0)
+
+    // MARK: - Views
+
+    private lazy var cartonsLabel = configureSectionLabel(titled: "Cartons")
+    private lazy var dateLabel = configureSectionLabel(titled: "Pickup Date")
+    private lazy var notesLabel = configureSectionLabel(titled: "Notes")
+
+    private lazy var tableView = configure(UITableView()) {
+        $0.register(PickupCartonCell.self,
+                    forCellReuseIdentifier: PickupCartonCell.reuseIdentifier)
+        $0.delegate = self
+    }
+    private lazy var dataSource = DataSource(
+        tableView: tableView,
+        cellProvider: cell(for:at:with:))
+    private lazy var addCartonButton = configure(UIButton()) {
```

Comment:
This configure function is one of my favorite takeaways from this project so far, lol. It is getting so much use.